src/ButtonGroup/index.js

```diff
@@ -0,0 +1,149 @@
+import React, { PureComponent, Children, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { transparentize } from 'polished';
+
+import Button from '../Button';
+
+const { func, node, string } = PropTypes;
+
+/**
+ * Button Group
+ *
+ * This component is in charge of displaying
+ * a button group
+ *
+ * @param {node} children // Anything that can be rendered: numbers, strings, elements or an array (or fragment).
+ * @param {string} className // Add a text aside in the select next the selected value.
+ * @param {function} onChange // Callback function called when a button is clicked.
+ * @param {string} defaultActiveButton // Specifies the initial state: witch button is selected.
+ *
+ * @return {jsx}
+ */
+
+class ButtonGroup extends PureComponent {
+  /**
+   * PropTypes Validation.
+   */
+  static propTypes = {
+    children: node,
+    className: string,
+    onChange: func,
+    defaultActiveButton: string,
+  };
+
+  /**
+   * Default props.
+   */
+  static defaultProps = {
+    children: null,
+    className: '',
+    onChange: () => {},
+    defaultActiveButton: '',
+  };
+
+  /** Internal state. */
+  state = {
+    activeButton: this.props.defaultActiveButton, // eslint-disable-line react/destructuring-assignment
+  };
+
+  /**
+   * updateActiveButton.
+   * Tell to the store witch button is activated
+   * @param {string} name
+   */
+  updateActiveButton = (name, hasClick) => {
+    const { onChange } = this.props;
+
+    this.setState({ activeButton: name }, () => {
+      onChange && onChange(name);
+      hasClick && hasClick();
+    });
+  };
+
+  /**
+   * Renders the component.
+   *
+   * @return {jsx}
+   */
+  render() {
+    const { children, className } = this.props;
+    const { activeButton } = this.state;
+
+    return (
+      <div role="group" className={className} tabIndex="-1">
+        {Children.map(children, child => {
+          let hasClick;
+          if (child.props.disabled) {
+            return child;
+          }
+          if (child.type !== Button) {
+            return child;
+          }
+          if (!child.props.name) {
+            throw new Error('"name" prop must be provided to Button');
+          }
+          if (child.props.onClick) {
+            hasClick = child.props.onClick;
+          }
+          return cloneElement(child, {
+            onClick: () => this.updateActiveButton(child.props.name, hasClick),
```

Comment:
Why not directly passing `child.props.onClick` here and checking whether it's set or not in `updateActiveButton`?
You'd gain in a readability I think.